packages/nouns-contracts/contracts/NounsArt.sol

```diff
@@ -0,0 +1,449 @@
+     * @notice Get the Trait struct for bodies.
+     * @dev This explicit getter is needed because implicit getters for structs aren't fully supported yet:
+     * https://github.com/ethereum/solidity/issues/11826
+     * @return Trait the struct, including a total image count, and an array of storage pages.
+     */
+    function getBodiesTrait() external view override returns (Trait memory) {
+        return bodiesTrait;
+    }
+
+    /**
+     * @notice Get the Trait struct for accessories.
+     * @dev This explicit getter is needed because implicit getters for structs aren't fully supported yet:
+     * https://github.com/ethereum/solidity/issues/11826
+     * @return Trait the struct, including a total image count, and an array of storage pages.
+     */
+    function getAccessoriesTrait() external view override returns (Trait memory) {
+        return accessoriesTrait;
+    }
+
+    /**
+     * @notice Get the Trait struct for heads.
+     * @dev This explicit getter is needed because implicit getters for structs aren't fully supported yet:
+     * https://github.com/ethereum/solidity/issues/11826
+     * @return Trait the struct, including a total image count, and an array of storage pages.
+     */
+    function getHeadsTrait() external view override returns (Trait memory) {
+        return headsTrait;
+    }
+
+    /**
+     * @notice Get the Trait struct for glasses.
+     * @dev This explicit getter is needed because implicit getters for structs aren't fully supported yet:
+     * https://github.com/ethereum/solidity/issues/11826
+     * @return Trait the struct, including a total image count, and an array of storage pages.
+     */
+    function getGlassesTrait() external view override returns (Trait memory) {
+        return glassesTrait;
+    }
+
+    /**
+     * @notice Batch add Noun backgrounds.
+     * @dev This function can only be called by the descriptor.
+     */
+    function addManyBackgrounds(string[] calldata _backgrounds) external override onlyDescriptor {
+        for (uint256 i = 0; i < _backgrounds.length; i++) {
+            _addBackground(_backgrounds[i]);
+        }
+
+        emit BackgroundsAdded(_backgrounds.length);
+    }
+
+    /**
+     * @notice Add a Noun background.
+     * @dev This function can only be called by the descriptor.
+     */
+    function addBackground(string calldata _background) external override onlyDescriptor {
+        _addBackground(_background);
+
+        emit BackgroundsAdded(1);
+    }
+
+    /**
+     * @notice Update a single color palette. This function can be used to
+     * add a new color palette or update an existing palette.
+     * @param paletteIndex the identifier of this palette
+     * @param palette byte array of colors. every 3 bytes represent an RGB color. max length: 256 * 3 = 768
+     * @dev This function can only be called by the descriptor.
+     */
+    function setPalette(uint8 paletteIndex, bytes calldata palette) external override onlyDescriptor {
```

Comment:
Thoughts on an additional function called `setPalettePointer`, which allows us to pre-deploy palettes?

Reply:
sounds good. will add 👍

Reply:
added https://github.com/nounsDAO/nouns-monorepo/pull/499/commits/8b16ca7f9735789d43ffbb753cb7b1465a9266f9